### ISSUE TITLE
vllm加速后会影响大模型的输出准确度吗？

### ISSUE THREAD

**Issue by PeterXiaTian** [vllm-project/vllm#2657]
如题．一般加速框架都会或多或少影响最终的准确度，请问vllm会吗？

**Comment by simon-mo** [comment]
No. Quantization is purely optional in vLLM. vLLM runs the model exactly as is and improve its throughput by better scheduling and memory management.

**Comment by PeterXiaTian** [comment]
> No. Quantization is purely optional in vLLM. vLLM runs the model exactly as is and improve its throughput by better scheduling and memory management.

well. But in fact ,when I try glm3-6b and glm3-6b with vllm, same prompt can output different result? why ?

**Comment by simon-mo** [comment]
vLLM and Huggingface model implementation should not differ by too much. The difference itself is commonly caused by greedy sampling and small numerical differences. We observe almost no difference on float32. 

Additionally, both implementations are "correct" because they use exactly the same model weight. 

Finally, seeding is also required to ensure the same output. 

You can check out the file in tests/models to see the check we are performing for correctness.

**Comment by PeterXiaTian** [comment]
> vLLM and Huggingface model implementation should not differ by too much. The difference itself is commonly caused by greedy sampling and small numerical differences. We observe almost no difference on float32.
> 
> Additionally, both implementations are "correct" because they use exactly the same model weight.
> 
> Finally, seeding is also required to ensure the same output.
> 
> You can check out the file in tests/models to see the check we are performing for correctness.

Does the prompt need to change? Are there any instructions on the prompt?

**Comment by simon-mo** [comment]
No you don't need to change the prompt.

**Comment by simon-mo** [comment]
No you don't need to change them All the parameters are interpreted the same way as huggingface implementation.

**Comment by PeterXiaTian** [comment]
> No you don't need to change them All the parameters are interpreted the same way as huggingface implementation.

ok.thanks.

well ,last problem,how to solve the output without prompt?　 
＂echo=false ＂　not taking effect　with the api  vllm/entrypoints/api_server.py

**Comment by simon-mo** [comment]
Try OpenAI compatible server's completions API.
